File: BACK-END/KuentecoChat/src/main/java/com/example/back_end/dto/response/ai/ReportDownloadResponseDTO.java
package com.example.back_end.dto.response.ai;

import com.example.back_end.enums.ResponseType;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
@EqualsAndHashCode(callSuper = true)
public class ReportDownloadResponseDTO extends BaseDynamicResponseDTO {
    private String reportId;
    private String fileName;
    private String reportType; // PDF o EXCEL
    private String reportCategory; // BALANCE, DEBT, SPENDING, etc.
    private long fileSizeBytes;
    private String expirationDate;

    public ReportDownloadResponseDTO(
            String summary,
            String analysis,
            String reportId,
            String fileName,
            String reportType,
            String reportCategory,
            long fileSizeBytes,
            String expirationDate) {
        super();
        this.setType(ResponseType.REPORT_DOWNLOAD);
        this.setSummary(summary);
        this.setAnalysis(analysis);
        this.reportId = reportId;
        this.fileName = fileName;
        this.reportType = reportType;
        this.reportCategory = reportCategory;
        this.fileSizeBytes = fileSizeBytes;
        this.expirationDate = expirationDate;
    }
}

File: FRONT-END/lib/dto/chat/response/report_download_response_dto.dart
class ReportDownloadResponseDTO {
  final String reportId;
  final String fileName;
  final String reportType;
  final String reportCategory;
  final int fileSizeBytes;
  final String expirationDate;

  ReportDownloadResponseDTO({
    required this.reportId,
    required this.fileName,
    required this.reportType,
    required this.reportCategory,
    required this.fileSizeBytes,
    required this.expirationDate,
  });

  factory ReportDownloadResponseDTO.fromJson(Map<String, dynamic> json) {
    return ReportDownloadResponseDTO(
      reportId: json['reportId'] as String,
      fileName: json['fileName'] as String,
      reportType: json['reportType'] as String,
      reportCategory: json['reportCategory'] as String,
      fileSizeBytes: json['fileSizeBytes'] as int,
      expirationDate: json['expirationDate'] as String,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'reportId': reportId,
      'fileName': fileName,
      'reportType': reportType,
      'reportCategory': reportCategory,
      'fileSizeBytes': fileSizeBytes,
      'expirationDate': expirationDate,
    };
  }
}